.github/workflows/release.yml

```diff
@@ -61,17 +57,151 @@ jobs:
 
   release-ci:
     name: Prerelease checks
+    if: inputs.dry_run == false
     needs:
     - acquire-base-image
     uses: ./.github/workflows/ci.yml
     with:
       run_sdk_examples: false
+      git_ref: ${{ inputs.commit_sha }}
+
+  get-or-create-release-branch:
+    name: Get or create a release branch
+    needs:
+      - release-ci
+    outputs:
+      release_branch: ${{ steps.branch-push.outputs.release_branch }}
+      new_release_series: ${{ steps.branch-push.outputs.new_release_series }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.commit_sha }}
+          token: ${{ secrets.RELEASE_AUTOMATION_BOT_PAT }}
+      - name: Get or create release branch
+        id: branch-push
+        shell: bash
+        env:
+          SEMANTIC_VERSION: ${{ inputs.semantic_version }}
+          DRY_RUN: ${{ inputs.dry_run }}
+        run: |
```

Comment:
Consider pulling this out to a separate shell script file so that it can be easily tested locally

Reply:
Done in [dd472e4](https://github.com/awslabs/smithy-rs/pull/2253/commits/dd472e435caa69cc15f378c16bc46c964c9471e2)
Shellcheck found _a lot_ of issues 😅 Thanks for pointing it out.